### PR TITLE
Feature/vault 0.9.1 version bump

### PR DESCRIPTION
**Pull request by mitch-roblox**

### CODE REVIEW

**Comment by mitch-roblox** [comment]
This combines an unmerged 3rd party PR: https://github.com/johnbellone/vault-cookbook/pull/102  with my own changes to bump the supported version up to 0.9.1 and add support for some HA options.

File: metadata.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@
 long_description 'Application cookbook for installing and configuring Vault.'
 issues_url 'https://github.com/johnbellone/vault-cookbook/issues'
 source_url 'https://github.com/johnbellone/vault-cookbook/'
-version '2.5.0'
+version '9002.0.0'
```
Comment (Brantone) [inline metadata.rb]: Should the version after "9000" be what the new version would be? ex: `9002.6.0`?

File: test/integration/default/inspec/default_spec.rb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-describe file('/opt/vault/0.6.5/vault') do
+describe file('/opt/vault/0.9.1/vault') do
```
Comment (Brantone) [inline test/integration/default/inspec/default_spec.rb]: Worth using the attribute `node['hashicorp-vault']['version']` ?

**Comment by Brantone** [comment]
I know some of the changes are expanding on other PR you mentioned, wasn't sure if worth tweaking it properly (ex: since could be maintainer), or just quickie "out the door" type bump.

**Comment by robloxrob** [comment]
Can you use node attributes in inspect tests?

**Comment by Brantone** [comment]
Nope, never mind, somehow read it as a file resource from a template. Comment removed.

**Comment by robloxrob** [comment]
It is a real bummer because that would be an excellent feature.

File: metadata.rb (hunk 1 above)
Comment (mitch-roblox) [reply metadata.rb]: It's a major version change, keeping the minor and patch versions would get messy, no?

**Comment by mitch-roblox** [comment]
You technically can access node attributes from InSpec but it's hacky af. And basically the maintainers of InSpec advise against it because integration tests should be more high-level.

**Comment by mitch-roblox** [comment]
Oh, and @Brantone 90% of this is from the other guy's PR, so I think there's any reason to mess with what he wrote unless I have to. Once his PR is merged, I'll rebase mine and submit it as a new PR to the upstream cookbook.

**Comment by Brantone** [comment]
Yeah, I looked up the PR you mentioned, wasn't sure if wanted to correct anything in the process or just subsequent changes. All good 👍

**Comment by mitch-roblox** [comment]
I might do a bit of tidying-up once the other PR is merged but I don't want to invest much time in that now.

File: metadata.rb (hunk 1 above)
Comment (Brantone) [reply metadata.rb]: Worth implementing new versioning scheme, or wait until confirmed?

Comment (mitch-roblox) [reply metadata.rb]: Yeah, I'll update with new version.

Comment (mitch-roblox) [reply metadata.rb]: Ok, I've pushed an updated version.

**Comment by Brantone** [comment]
LGTM 👍

**Pull request merged**